src/app/search/SearchFilter.tsx

```diff
@@ -133,9 +136,34 @@ class SearchFilter extends React.Component<Props, State> {
     }
   );
 
+  private compareMatching = loadingTracker.trackPromise(async () => {
+    const comparableItems = this.getStoresService()
+      .getAllItems()
+      .filter(this.props.searchFilter);
+    CompareService.addMultiItemsToCompare(comparableItems);
+  });
+
+  onQueryChanged(query: string): void {
+    this.props.setSearchQuery(query);
+    loadingTracker.trackPromise(async () => {
+      const filteredItems = this.getStoresService()
+        .getAllItems()
+        .filter(this.props.searchFilter);
+
+      if (!filteredItems.length) {
+        return;
+      }
+      const compareType = CompareService.compareType;
+      const type = filteredItems[0].typeName;
+      const isComparable = !filteredItems.some((i) => i.typeName !== (compareType || type));
```

Comment:
I'd phrase this as `filteredItems.every((i) => i.typeName === type)`.

Reply:
Wanted to break early using `some`, unless you think `every` reads better?

Reply:
It'll still break early if the predicate returns false.